### PR TITLE
Use ManagedResources to deploy ControllerInstallations

### LINKED ISSUE #1574: Use central gardener-resource-manager in seeds to deploy extensions
Since https://github.com/gardener/gardener/pull/1331 every seed gets a central `gardener-resource-manager` deployment. We should use this one and its `ManagedResource` API to deploy and reconcile the extension controllers instead of duplicating this logic in the `ControllerInstallation` controller: https://github.com/gardener/gardener/blob/master/pkg/controllermanager/controller/controllerinstallation/controllerinstallation_control.go#L147

### PR DESCRIPTION
**Pull request by ialidzhikov**
**What this PR does / why we need it**:
Use ManagedResources to deploy ControllerInstallations

**Which issue(s) this PR fixes**:
Fixes #1574

**Special notes for your reviewer**:

**Release note**:

```improvement operator
ControllerInstallation controller does now uses ManagedResources.
```

### CODE REVIEW

File: docs/extensions/controllerregistration.md [hunk 1]
```diff
@@ -112,15 +112,13 @@ status:
     type: Installed
 ```
 
-Additionally, the `.status` field has a `providerStatus` section into which the operator can (optionally) put any arbitrary data associated with this installation.
```
Comment (rfranzke) [inline docs/extensions/controllerregistration.md]: Revert this deletion

File: pkg/apis/core/v1alpha1/types_controllerinstallation.go [hunk 2]
```diff
@@ -60,6 +60,7 @@ type ControllerInstallationStatus struct {
 	// +optional
 	Conditions []Condition `json:"conditions,omitempty"`
 	// ProviderStatus contains type-specific status.
+	// DEPRECATED: This field will be removed in a future version.
```
Comment (rfranzke) [inline pkg/apis/core/v1alpha1/types_controllerinstallation.go]: Why do you want to remove this? We have to keep it as we don't know who installs the extension. Gardener implements a standard installer for `helm`-typed registrations, but there might be other types, and here the responsible installer might want to store status in the `providerStatus` field. Please revert this or explain why you think it's no longer needed.

File: pkg/operation/botanist/resource_manager.go [hunk 3]
```diff
@@ -62,7 +62,9 @@ func (b *Botanist) DeleteManagedResources(ctx context.Context) error {
 func (b *Botanist) WaitUntilManagedResourcesDeleted(ctx context.Context) error {
 	return retry.Until(ctx, 5*time.Second, func(ctx context.Context) (done bool, err error) {
 		managedResources := &resourcesv1alpha1.ManagedResourceList{}
-		if err := b.K8sSeedClient.Client().List(ctx, managedResources, client.InNamespace(b.Shoot.SeedNamespace), client.MatchingLabels(map[string]string{ManagedResourceLabelKeyOrigin: ManagedResourceLabelValueGardener})); err != nil {
+		if err := b.K8sSeedClient.Client().List(ctx, managedResources,
```
Comment (rfranzke) [inline pkg/operation/botanist/resource_manager.go]:
```suggestion
		if err := b.K8sSeedClient.Client().List(ctx,
		managedResources,
```

If you want to have a newline for the arguments then do it for every argument :)

File: pkg/controllermanager/controller/controllerinstallation/controllerinstallation_control.go [hunk 4]
```diff
@@ -488,6 +471,9 @@ func (c *defaultControllerInstallationControl) cleanOldExtensions(ctx context.Co
 	return nil
 }
 
+// cleanOldResources cleans ControllerInstallation resources which are not found in the given <newResourcesSet>.
+//
+// TODO: Kept for backwards compatibility. Remove after several releases.
```
Comment (rfranzke) [inline pkg/controllermanager/controller/controllerinstallation/controllerinstallation_control.go:511]: Nice! Should we open a separate issue to keep track? We haven't done that in the past, but I feel that the number of such things is increasing and no longer be handleable without issues?

File: pkg/controllermanager/controller/controllerinstallation/controllerinstallation_control.go [hunk 5]
```diff
@@ -261,67 +261,30 @@ func (c *defaultControllerInstallationControl) reconcile(controllerInstallation
...
+		data[key] = []byte(fileContent)
 	}
 
-	if err := c.cleanOldResources(k8sSeedClient, controllerInstallation, newResourcesSet); err != nil {
-		if isDeletionInProgressError(err) {
-			conditionInstalled = helper.UpdatedCondition(conditionInstalled, gardencorev1alpha1.ConditionFalse, "DeletionPending", err.Error())
-		} else {
-			conditionInstalled = helper.UpdatedCondition(conditionInstalled, gardencorev1alpha1.ConditionFalse, "DeletionFailed", fmt.Sprintf("Deletion of old resources failed: %+v", err))
-		}
+	secretName := "managedresource-" + controllerInstallation.Name
```
Comment (rfranzke) [inline pkg/controllermanager/controller/controllerinstallation/controllerinstallation_control.go]: Why does this have the `managedresource-` prefix and the MR itself does not?
Should we prefix both with `controllerinstallation-<installationName>` or simply keep the `<installationName>` as name for both secret and MR?

**Review by rfranzke — CHANGES_REQUESTED** [review]

**Comment by ialidzhikov** [comment]
Thanks for the review. Let's target it for `0.33.0` as the guys are now validating `0.32.0`.

File: pkg/controllermanager/controller/controllerinstallation/controllerinstallation_control.go (hunk 4 above)
Comment (ialidzhikov) [reply pkg/controllermanager/controller/controllerinstallation/controllerinstallation_control.go:511]: Agree, we can open one.

File: pkg/controllermanager/controller/controllerinstallation/controllerinstallation_control.go (hunk 5 above)
Comment (ialidzhikov) [reply pkg/controllermanager/controller/controllerinstallation/controllerinstallation_control.go]: I changed it to `<installationName>` for both mr and secret name.

**Review by rfranzke — APPROVED** [review]
/lgtm

**Pull request merged**